migrate/20250227_id_based_locations.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  up do
+    alter_table(:vm_host) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:vm) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:vm_pool) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:private_subnet) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:firewall) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:postgres_resource) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:minio_cluster) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:inference_endpoint) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    alter_table(:kubernetes_cluster) do
+      add_foreign_key :location_id, :location, type: :uuid
+    end
+
+    run "UPDATE vm_host SET location_id = (SELECT id FROM location WHERE name = vm_host.location);"
+    run "UPDATE vm SET location_id = (SELECT id FROM location WHERE name = vm.location);"
+    run "UPDATE vm_pool SET location_id = (SELECT id FROM location WHERE name = vm_pool.location);"
+    run "UPDATE private_subnet SET location_id = (SELECT id FROM location WHERE name = private_subnet.location);"
+    run "UPDATE firewall SET location_id = (SELECT id FROM location WHERE name = firewall.location);"
+    run "UPDATE postgres_resource SET location_id = (SELECT id FROM location WHERE name = postgres_resource.location);"
+    run "UPDATE minio_cluster SET location_id = (SELECT id FROM location WHERE name = minio_cluster.location);"
+    run "UPDATE inference_endpoint SET location_id = (SELECT id FROM location WHERE name = inference_endpoint.location);"
+    run "UPDATE kubernetes_cluster SET location_id = (SELECT id FROM location WHERE name = kubernetes_cluster.location);"
+
+    alter_table(:vm_host) do
+      drop_column :location
```

Comment:
It's too dangerous to drop the location columns in the same commit, because the migration will be run before the code changes that use it are submitted.  This needs to be staged, with the location column dropped in a later migration.  You'll need to make sure that the tests pass at every stage:

* migration that adds location_id
* Commit that switches from using location to using location_id
* migration that drops location

You probably want to mark the `location` columns as allowing `NULL` values with `set_column_allow_null`, so that later commit can stop setting them, and a following migration can drop them.

Reply:
Updated accordingly